test/suites/test_dml.py

```diff
@@ -302,6 +357,51 @@ def test_12_update_fields(self):
             [[2, 'help', 7]]
         )
 
+    def test_13_00_string_insert_encoding_utf8_behavior(self):
+        self.assertNotRaises(
+            self.con_encoding_utf8.insert,
+            'space_str', [ 'test_13_00' ])
+
+    def test_13_01_string_select_encoding_utf8_behavior(self):
+        self.adm(r"box.space['space_str']:insert{'test_13_01'}")
+
+        strdata = 'test_13_01'
+        resp = self.con_encoding_utf8.select('space_str', [strdata])
+        self.assertEquals(resp[0][0], strdata)
+
+    @skip_or_run_mp_bin_test
+    @skip_or_run_varbinary_test
+    def test_13_02_varbinary_insert_encoding_utf8_behavior(self):
+        self.assertNotRaises(
+            self.con_encoding_utf8.insert,
+            'space_varbin', [ b'test_13_02' ])
+
+    @skip_or_run_mp_bin_test
+    @skip_or_run_varbinary_test
+    def test_13_03_varbinary_select_encoding_utf8_behavior(self):
+        self.adm(r"""
+        bintuple_insert(
+            box.space['space_varbin'],
+            {0xDE, 0xAD, 0xBE, 0xAF, 0x13, 0x03})
+        """)
+
+        bindata = bytes(bytearray.fromhex('DEADBEAF1303'))
+        resp = self.con_encoding_utf8.select('space_varbin', [bindata])
+        self.assertEquals(resp[0][0], bindata)
+
+    def test_14_00_string_insert_encoding_none_behavior(self):
```

Comment:
`encoding=None` testing looks incomplete:

* OK: bytes -> mp_str encoding
* OK: mp_str -> bytes decoding
* NO: str -> ? encoding
* NO: mp_bin -> ? decoding

As for me, completeness/orthogonality will make the test easier to read.

It would also be good to mention actual behaviour of encoding `str` with `encoding=None` (and decoding `mp_bin` with `encoding=None`) in the PR description aside of the fact that it is not changed.

Reply:
Added tests and description for old behavior and new behavior for all cases

Reply:
Nice, thanks!